src/com/parse4cn1/command/ParseUploadCommand.java

```diff
@@ -54,6 +54,7 @@ void setUpRequest(ConnectionRequest request) throws ParseException {
 
         if (contentType != null) {
             request.addRequestHeader(ParseConstants.HEADER_CONTENT_TYPE, contentType);
+            addHeader(ParseConstants.HEADER_CONTENT_TYPE, contentType);
```

Comment:
Yeah i think that would be better.
But we cant keep track of all the headers because there is no getters for the `userHeaders `in the `ConnectionRequest `class so unless we keep track of it using a local variable it want work, 
However there is a getter for contentType, So we could use that with the content type or we could get ride of assigning headers to the request and use addHeaders only .. what you think?